src/DeclarativePackages.jl

```diff
@@ -1,5 +1,27 @@
 module DeclarativePackages
 
+if VERSION < v"0.5.0-dev+2228"
+    const readstring = readall
+    export readstring
+end
+
+if VERSION < v"0.5.0-dev+4267"
+    if OS_NAME == :Windows
+        const KERNEL = :NT
+    else
+        const KERNEL = OS_NAME
+    end
+
+    @eval is_apple()   = $(KERNEL == :Darwin)
+    @eval is_linux()   = $(KERNEL == :Linux)
+    @eval is_bsd()     = $(KERNEL in (:FreeBSD, :OpenBSD, :NetBSD, :Darwin, :Apple))
+    @eval is_unix()    = $(is_linux() || is_bsd())
+    @eval is_windows() = $(KERNEL == :NT)
+    export is_apple, is_linux, is_bsd, is_unix, is_windows
+else
+    const KERNEL = Sys.KERNEL
+end
+
```

Comment:
I think we initially wanted to drop support for anything before 0.5 - not sure if there would be a disadvantage in using Compat.jl